File: src/api/game.ts
import { invoke } from "@tauri-apps/api/core";
import type { GameTextLanguage } from "../domain/gameText";

export interface GameDirectoryStatus {
  path: string | null;
  isConfigured: boolean;
  isValid: boolean;
  message: string;
  executablePath: string | null;
  nativePcPath: string | null;
  configPath: string;
  source: string;
}

export interface GameTextSettings {
  language: GameTextLanguage;
}

export function getGameDirectoryStatus(): Promise<GameDirectoryStatus> {
  return invoke<GameDirectoryStatus>("get_game_directory_status");
}

export function detectGameDirectory(): Promise<GameDirectoryStatus> {
  return invoke<GameDirectoryStatus>("detect_game_directory");
}

export function saveGameDirectory(path: string): Promise<GameDirectoryStatus> {
  return invoke<GameDirectoryStatus>("save_game_directory", { path });
}

export function getGameTextSettings(): Promise<GameTextSettings> {
  return invoke<GameTextSettings>("get_game_text_settings");
}

export function saveGameTextLanguage(language: GameTextLanguage): Promise<GameTextSettings> {
  return invoke<GameTextSettings>("save_game_text_language", { language });
}

File: src-tauri/src/commands/game.rs
use crate::{
    services::game::{self, GameDirectoryStatus, GameTextSettings},
    storage::config::GameTextLanguage,
};

#[tauri::command]
pub fn get_game_directory_status(app: tauri::AppHandle) -> Result<GameDirectoryStatus, String> {
    game::get_game_directory_status(&app)
}

#[tauri::command]
pub fn detect_game_directory(app: tauri::AppHandle) -> Result<GameDirectoryStatus, String> {
    game::detect_game_directory(&app)
}

#[tauri::command]
pub fn save_game_directory(
    app: tauri::AppHandle,
    path: String,
) -> Result<GameDirectoryStatus, String> {
    game::save_game_directory(&app, path)
}

/// 读取武器、防具等游戏内容名称使用的语言。
#[tauri::command]
pub fn get_game_text_settings(app: tauri::AppHandle) -> Result<GameTextSettings, String> {
    game::get_game_text_settings(&app)
}

/// 保存游戏内容名称语言；该设置不改变 Acumod 自身界面语言。
#[tauri::command]
pub fn save_game_text_language(
    app: tauri::AppHandle,
    language: GameTextLanguage,
) -> Result<GameTextSettings, String> {
    game::save_game_text_language(&app, language)
}
